lib/segment/src/types.rs

```diff
@@ -418,6 +418,17 @@ pub struct ScalarQuantizationConfig {
     pub always_ram: Option<bool>,
 }
 
+impl ScalarQuantizationConfig {
+    /// Detect configuration mismatch against `other` that requires rebuilding
+    ///
+    /// Returns true only if both conditions are met:
```

Comment:
The comment says both conditions are required but the code using an OR.
It seems one condition is sufficient to trigger a rebuild.

Reply:
On more careful look, the doc is correct. It does not refer to the OR statement.

The first condition describes that fields differ. The second condition describes that, to fix that, we have to rebuilt.

We must match both of those conditions.

Reply:
Yes, right now the second condition is always true as an effect of the first condition. In other words, right now the first condition implies the second. But this won't be the case in the future.

Here is an example. In the future we don't have to rebuilt a whole segment for just flipping the `always_ram` parameter. In that case, the first condition would be true (different parameters), but the second condition would be falls (don't require rebuilt).

Note that for now, we do rebuilt even if flipping this parameter because we don't have logic yet to swap between RAM and on-disk on the fly.